lit/docs/guides/pipeline-guides.lit

```diff
@@ -0,0 +1,676 @@
+    resources:
+    - name: ci
+      type: git
+      source:
+        uri: http://github.com/username/my-ci-repo
+
+    jobs:
+    - name: the-best-job
+      plan:
+      - get: ci
+      - task: what-day-is-it
+        file: ci/tasks/todays-date.yml
+    }}
+  }
+
+  \section{
+    \title{\code{Get} Images for Tasks Instead of using Anonymous Image Resources}
+
+    It is easy to let Concourse fetch images for tasks right when they are
+    needed, it's the easy out-of-the-box solution. Another way is to pass the
+    image(s) for tasks as inputs to jobs.
+
+    \bold{Before - Anonymous Image Fetching for Tasks}
+
+    \codeblock{yaml}{{
+    jobs:
+    - name: job
+      plan:
+      - task: simple-task
+        config:
+          platform: linux
+          image_resource: # anonymous image resource
+            type: registry-image
+            source:
+              repository: busybox
+          run:
+            path: echo
+            args: ["Hello world!"]
+    }}
+
+    \inset{} {- spacer -}
+
+    \bold{After - Passing Task Image as Job Inputs}
+
+    \codeblock{yaml}{{
+    resources:
+    - name: busybox
+      type: registry-image
+      source:
+        repository: busybox
+        username: ((docker.user))
+        password: ((docker.password))
+
+    jobs:
+    - name: job
+      plan:
+      - get: busybox # pass image into job
+      - task: simple-task
+        image: busybox # use image for task. Overrides anonymous image
+        config:
+          platform: linux
+          image_resource: # overridden by step-level "image" field
+            type: registry-image
+            source:
+              repository: busybox
+          run:
+            path: echo
+            args: ["Hello world!"]
+    }}
+  }
+}
+
+\section{
+  \omit-children-from-table-of-contents
+  \title{Exploring Task Input and Output Scenarios}
+  Understanding how task inputs and outputs work in Concourse can be a little
+  confusing initially. This guide will walk you through a few example pipelines
+  to show you how inputs and outputs work within a single Concourse job. By the
+  end you should understand how inputs and outputs work within the context of a
+  single job.
+
+  To run the pipelines in the following examples yourself you can get your own
+  Concourse running locally by following the \reference{quick-start} guide.
+  Then use \reference{fly-set-pipeline} to see the pipelines in action.
+
+  \section{
+    \title{#1 - Passing Inputs Between Tasks}
+
+    This pipeline will show us how to create outputs and pass outputs as inputs
+    to the next \reference{steps}{step} in a \reference{schema.job.plan}{job plan}.
+
+    This pipeline has two tasks. The first task outputs a file with the date. The
+    second task reads and prints the contents of the file from the first task.
+
+    \titled-codeblock{passing-artifacts.yml}{yaml}{{
+    busybox: &busybox #YAML anchor
+      type: registry-image
+      source: {repository: busybox}
```

Comment:
the `{...}` get clobbered here:

<img width="306" alt="Screen Shot 2021-05-21 at 12 37 25 PM" src="https://user-images.githubusercontent.com/26583442/119170499-4d6bb400-ba31-11eb-87e9-2f2f50ef97f5.png">